Gems/Atom/Tools/MaterialCanvas/Assets/MaterialCanvas/GraphData/Nodes/MaterialOutputs/BasePBR/MaterialGraphName_SurfaceEval.azsli

```diff
@@ -50,7 +51,7 @@ Surface EvaluateSurface_MaterialGraphName(
     Surface surface;
     surface.position = positionWS;
     surface.vertexNormal = vertexNormal;
-    surface.normal = normalize(o_normal_override_enabled ? (float3)inNormal : vertexNormal);
+    surface.normal = normalize(o_normal_override_enabled ? GetWorldSpaceNormal((float2)inNormal, vertexNormal, tangents[0], bitangents[0]) : vertexNormal);
```

Comment:
Calling `GetWorldSpaceNormal` requires `inNormal` to be normalized (seems so in the pixel evaluation). It should be better to use `inNormal.xy` instead of casting.
I think you can directly use
`TangentSpaceToWorld(inNormal, vertexNormal, tangents[0], bitangents[0])`
So that no casting or normalization is needed.

Reply:
Thank you! I have to cast instead of using .xy because the generated variable could end up being a different type, a scalar that does not have .xy.